### DISCUSSION TITLE
Question about ipxe versioning

### DISCUSSION THREAD

**Discussion by kaleal** [ipxe/ipxe#544] [Q&A] [answered]
Hello,

I have seen several changes on ipxe over the latest year but the latest version (tag v1.21.1) has been released one year ago.
Are the users supposed to be using that version or the versioning isn't being used and users are expected to be using very latest commit?

If versioning isn't being used, wouldn't be better to start creating releases/tags after each change is fully validated?

**Comment by kaleal** [comment]
There are 197 commits after v1.21.1 was released.

![image](https://user-images.githubusercontent.com/15032576/147859907-060dff3c-36b5-4d8d-8b77-7c2693ece36a.png)

**Comment by NiKiZe** [comment, marked as answer]
Git Master (as linked from https://ipxe.org/download) is to be expected as current stable.
Prebuild binaries are hourly built and available from boot.ipxe.org

**Comment by kaleal** [comment]
Thanks for moving this to the right place.

The problem of getting code directly from master branch is because it can (and will) change. If I checkout the code twice on different days and compile them I could potentially have two different results. If I checkout from a tag instead I would have consistent results between builds.

Imagine the following scenario:
- I have a builder which checks out ipxe, apply my local configuration, and builds it.

If I change the configuration and run the builder again I will potentially build a different ipxe version together with my configuration. That would make it harder to troubleshoot if something goes wrong.

A solution for this would be creating a tag/release each time the code is validated (tests on cicd pipeline?). So, for the same scenario above my configuration could be applied to the exactly the same version if I run it in a few months interval. Also it would make it easier to upgrade to a newest ipxe version, since I would know if changing the version and keeping the configuration would break something else.

**Reply by NiKiZe** [reply]
If that is important, then use git hash.
Each commit to master is tested and should be seen as production ready.

Git version is also included in iPXE version, exactly to make it easy to debug.

**Comment by kaleal** [comment]
Releases can be created using github API. I could help with that.

**Reply by NiKiZe** [reply]
Yes it can, but if we are talking about binaries the latest is always available from boot.ipxe.org. if talking about sources, then each master commit can also easily be downloaded already.

I think the only reason there even is a 1.21 version is due to it being a hard requirement by one of the customers that paid to have such a release. Before that there was 1.0.0+ with git tag.

**Reply by stappersg** [reply]
Cross reference to #679

**Reply by stappersg** [reply]
Cross reference to #794

**Comment by Neustradamus** [comment]
@NiKiZe: Like I have done my personal ticket, I confirm that it is needed to have new official builds!
- https://github.com/ipxe/ipxe/issues/904

Linked to:
- https://github.com/ipxe/ipxe/discussions/887
- https://github.com/ipxe/ipxe/issues/886
- https://github.com/ipxe/ipxe/pull/794
- https://github.com/ipxe/ipxe/discussions/679
- https://github.com/ipxe/ipxe/discussions/544
